rtl/eclrtl/rtldynfield.cpp

```diff
@@ -510,3 +510,481 @@ extern ECLRTL_API int getFieldNum(const char *fieldName, IOutputMetaData &  meta
+                            }
+                        }
+                        else
+                        {
+                            // a 32-bit size, then rows inline
+                            size32_t childSize = *(size32_t *) source;
+                            source += sizeof(size32_t);
+                            const byte *initialSource = source;
+                            while (source - initialSource < childSize)
+                            {
+                                offset = match.subTrans->translate(builder, offset, source);
+                                source += sourceType->queryChildType()->size(source, nullptr); // MORE - shame to repeat a calculation that the translate above almost certainly just did
+                            }
+                        }
+                        dest = builder.getSelf() + countOffset;  // Note - man have been moved by reallocs since last calculated
+                        *(size32_t *)dest = offset - initialOffset;
+                    }
+                    break;
+                default:
+                    throwUnexpected();
+                }
+            }
+            //printf("after idx %d offset %d\n", idx, offset);
+        }
+        return offset;
+    }
+    inline FieldMatchType match() const
+    {
+        return matchFlags;
+    }
+    bool canTranslate() const
+    {
+        return (matchFlags & match_fail) == 0;
+    }
+    bool needsTranslate() const
+    {
+        return (matchFlags & ~match_link) != 0;
+    }
+private:
+    const RtlRecord &destRecInfo;
+    const RtlRecord &sourceRecInfo;
+    FieldMatchType matchFlags = match_perfect;
+
+    struct MatchInfo
+    {
+        unsigned matchIdx = 0;
+        FieldMatchType matchType = match_fail;
+        char fillChar = 0;
+        GeneralRecordTranslator *subTrans = nullptr;
+        ~MatchInfo()
+        {
+            delete subTrans;
+        }
+    } *matchInfo;
+
+    static size32_t translateScalar(ARowBuilder &builder, size32_t offset, const RtlFieldInfo *field, const RtlTypeInfo *destType, const RtlTypeInfo *sourceType, const byte *source)
+    {
+        // This code COULD move into rtlfield.cpp?
+        switch(destType->getType())
+        {
+        case type_boolean:
+        case type_int:
+        case type_swapint:
+        case type_packedint:
+            offset = destType->buildInt(builder, offset, field, sourceType->getInt(source));
+            break;
+        case type_real:
+            offset = destType->buildReal(builder, offset, field, sourceType->getReal(source));
+            break;
+        case type_decimal:
+        case type_data:
+        case type_string:
+        case type_varstring:
+        case type_qstring:
+        {
+            size32_t size;
+            rtlDataAttr text;
+            sourceType->getString(size, text.refstr(), source);
+            offset = destType->buildString(builder, offset, field, size, text.getstr());
```

Comment:
I think ebcdic should be ok - it's handled at the far end. Ebcdic strings may und up converted to ascii and back as part of the translation, but that's tough.